source/Engine.cpp

```diff
@@ -2046,14 +2044,17 @@ void Engine::HandleMouseInput(Command &activeCommands)
 {
 	isMouseHoldEnabled = activeCommands.Has(Command::MOUSE_TURNING_HOLD);
 	if(activeCommands.Has(Command::MOUSE_TURNING_TOGGLE))
+	{
 		isMouseToggleEnabled = !isMouseToggleEnabled;
+		Preferences::Set("alt-mouse turning", isMouseTurningEnabled);
```

Comment:
This is unnecessary now since the preference is never read anymore.

Reply:
This is still necessary for right alt toggle.  This PR only addresses alt+f4 exit bug.

Terin mentioned they wanted to do a preference fix for this so I limited it to fixing the alt+f4 bug.

Reply:
This might be a non-persistence bug.  Toggle should preserve its state across restart.

Reply:
Yes this PR has a persistence bug.

The point of mouse toggle using a preference is the toggle persists across game restart.  So if you toggle mouse control, the next time you launch the game or reload the game it should have remembered the current preference (with regard to the toggle).

The problem statement this is fixing is left alt is a temporary toggle only as long as alt is held.  i.e. temporarily enable or disable mouse control.

If someone pressed alt+f4 (left alt) the game would exit but because the preference was used left alt would toggle on exit (the bug)